pkg/webhook/validatingwebhook/validate.go

```diff
@@ -0,0 +1,144 @@
+	defer func() {
+		if err := r.Body.Close(); err != nil {
+			log.Error(err, "unable to close the body")
+		}
+	}()
+	if err != nil {
+		log.Error(err, "unable to read the body of the request")
+		w.WriteHeader(http.StatusInternalServerError)
+		respBody = []byte(fmt.Sprintf("unable to read the body of the request: %s", err))
+	} else {
+		// validate the request
+		respBody = validate(body, v.Client)
+		w.WriteHeader(http.StatusOK)
+	}
+	if _, err := w.Write(respBody); err != nil {
+		log.Error(err, "unable to write response")
+	}
+}
+
+func validate(body []byte, client runtimeClient.Client) []byte {
+	admReview := v1.AdmissionReview{}
+	if _, _, err := deserializer.Decode(body, nil, &admReview); err != nil {
+		log.Error(err, "unable to deserialize the admission review object", "body", body)
+		return denyAdmissionRequest(admReview, errors.Wrapf(err, "unable to deserialize the admission review object - body: %v", body))
+	}
+	// let's unmarshal the object to be sure that it's a rolebinding
+	var rb *rbac.RoleBinding
+	if err := json.Unmarshal(admReview.Request.Object.Raw, &rb); err != nil {
+		log.Error(err, "unable unmarshal rolebinding json object", "AdmissionReview", admReview)
+		return denyAdmissionRequest(admReview, errors.Wrapf(err, "unable unmarshal rolebinding json object - raw request object: %v", admReview.Request.Object.Raw))
+	}
+	requestingUsername := admReview.Request.UserInfo.Username
+	// allow admission request if the user is a system user
+	if strings.HasPrefix(requestingUsername, "system:") {
+		return allowAdmissionRequest(admReview)
+	}
+	subjects := rb.Subjects
+	allServiceAccountsSubject := rbac.Subject{
+		Kind:     "Group",
+		Name:     "system:serviceaccounts",
+		APIGroup: "rbac.authorization.k8s.io",
+	}
+	allUsersSubject := rbac.Subject{
+		Kind:     "Group",
+		Name:     "system:authenticated",
+		APIGroup: "rbac.authorization.k8s.io",
+	}
+	for _, sub := range subjects {
+		if sub == allUsersSubject || sub == allServiceAccountsSubject {
+			requestingUser := &userv1.User{}
+			err := client.Get(context.TODO(), types.NamespacedName{
+				Name: admReview.Request.UserInfo.Username,
+			}, requestingUser)
+
+			if err != nil {
+				log.Error(fmt.Errorf("Cannot find the user: %w", err), "unable to find the user requesting creation")
+				return denyAdmissionRequest(admReview, errors.Wrapf(err, "unable to find the user requesting creation: %s", requestingUsername))
+			}
+			//check if the requesting user is a sandbox user
+			if requestingUser.GetLabels()[toolchainv1alpha1.ProviderLabelKey] == toolchainv1alpha1.ProviderLabelValue {
+				log.Info("trying to give access which is restricted", "unable unmarshal rolebinding json object", "AdmissionReview", admReview)
+				return denyAdmissionRequest(admReview, errors.Wrapf(fmt.Errorf("trying to give access which is restricted"), "Unauthorized request to create rolebinding json object - raw request object: %v", admReview.Request.Object.Raw))
+			}
+
```

Comment:
By this point you have checked the user and it's not our user. So we do not have to go over other subjects anymore. Just break the loop here.

Reply:
done in 9d1180e